planseqlearn/psl/robosuite_mp_env.py

```diff
@@ -775,29 +981,148 @@ def compute_hardcoded_orientation(self, target_pos, quat):
+                object_quat = T.convert_quat(self.sim.data.qpos[33:37].copy(), to="xyzw")
+            elif self.env_name.endswith("PickPlace"):
+                if self.text_plan[self.curr_plan_stage][1] == "grasp":
+                    all_obj_names = [name.lower() for name in ["Milk", "Bread", "Cereal", "Can"] if name in self.valid_obj_names]
+                    new_obj_idx = [name for name in enumerate(all_obj_names) if name[1] in obj_name][0][0]
+                    object_pos = self.sim.data.qpos[
+                        9 + 7 * new_obj_idx : 12 + 7 * new_obj_idx
+                    ].copy()
+                    object_quat = T.convert_quat(
+                        self.sim.data.qpos[12 + 7 * new_obj_idx : 16 + 7 * new_obj_idx].copy(),
+                        to="xyzw",
+                    )
+                elif self.text_plan[self.curr_plan_stage][1] == "place":
+                    assert self.text_plan[self.curr_plan_stage][0].lower().startswith("bin"), "placement location must be a bin"
+                    bin_num = int(self.text_plan[self.curr_plan_stage][0][-1])
+                    object_pos = self.pick_place_bin_locations[bin_num - 1].copy()
+                    object_pos[2] += 0.125
+                    object_quat = np.zeros(4)
+            elif self.env_name.startswith("Door"):
+                object_pos = np.array(
+                    [self.sim.data.qpos[self.hinge_qpos_addr]]
+                )  # this is not what they are, but they will be decoded properly
+                object_quat = np.array(
+                    [self.sim.data.qpos[self.handle_qpos_addr]]
+                )  # this is not what they are, but they will be decoded properly
+            elif "NutAssembly" in self.env_name:
+                if "grasp" in self.text_plan[self.curr_plan_stage][1]:
+                    if "gold" in obj_name:
+                        nut = self.nuts[0]
+                    elif "silver" in obj_name:
+                        nut = self.nuts[1]
+                    else:
+                        raise NotImplementedError
+                    object_pos = self.sim.data.get_site_xpos(nut.important_sites["handle"])
+                    object_quat = T.convert_quat(
+                        self.sim.data.body_xquat[self.obj_body_id[nut.name]], to="xyzw"
+                    )
+                elif "place" in self.text_plan[self.curr_plan_stage][1]:
+                    if "gold" in obj_name:
+                        object_pos = np.array(self.sim.data.body_xpos[self.peg1_body_id])
+                    elif "silver" in obj_name:
+                        object_pos = np.array(self.sim.data.body_xpos[self.peg2_body_id])
+                    object_pos[2] += 0.15
+                    object_pos[0] -= 0.065
+                    object_quat = np.zeros(4) # not used
+            else:
+                raise NotImplementedError
+        return object_pos, object_quat
+    
+    def named_get_object_pose(self, obj_name):
```

Comment:
for this sort of function, maybe specify its not using SAM? like get_oracle_pos (using sim) vs. estimated (using sam)